lucene/core/src/java/org/apache/lucene/codecs/lucene80/Lucene80DocValuesProducer.java

```diff
@@ -742,6 +755,131 @@ public BytesRef binaryValue() throws IOException {
         };
       }
     }
+  }  
+  
+  // Decompresses blocks of binary values to retrieve content
+  class BinaryDecoder {
+    
+    private final LongValues addresses;
+    private final IndexInput compressedData;
+    // Cache of last uncompressed block 
+    private long lastBlockId = -1;
+    private int []uncompressedDocEnds = new int[Lucene80DocValuesFormat.BINARY_DOCS_PER_COMPRESSED_BLOCK];
```

Comment:
in the past we've put these constants in the meta file and BinaryEntry so that it's easier to change values over time

Reply:
@jpountz we should use the same structure while writing the data, in that case you will see all the properties of the class instead of adding comments in the code